README.md

```diff
@@ -153,6 +154,100 @@ The business logic is designed to be purely functional and segregated from
+[service layer pattern](https://martinfowler.com/eaaCatalog/serviceLayer.html)
+and handles the business logic exposed through the API.
+
+The service layer is composed of several specialized controllers that handle
+operations for specific resources:
+
+|Controller|Responsibilities|
+|-|-|
+|[Messages](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/controllers/messages.ts)|Handles operations on the Message resource|
+|[Profiles](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/controllers/profiles.ts)|Handles operations on the Profile resource|
+|[Info](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/controllers/info.ts)|Provides system runtime information|
+|[OpenAPI](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/controllers/openapi.ts)|Provides the API specification in [OpenAPI](https://www.openapis.org/) format|
+
+The specialized controllers are designed to follow the [Zalando RESTful API guidelines](https://zalando.github.io/restful-api-guidelines/index.html).
+
+#### Data source layer
+
+The data source layer follows the
+[table data gateway pattern](https://martinfowler.com/eaaCatalog/tableDataGateway.html)
+and handles the operations on the resources stored in the database.
+
+Operations specific to the databased technology have been encapsulated in a
+[gateway object](https://martinfowler.com/eaaCatalog/gateway.html) that
+[abstracts the DocumentDb API](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/utils/documentdb_model.ts) exposed by CosmosDB. This approach decouples the
+underlying data store semantic from the application, making possible to
+migrate the application to a different data store with little effort.
+
+The data model is designed around several common techniques like
+[identity field](https://martinfowler.com/eaaCatalog/identityField.html),
+[foreign key mapping](https://martinfowler.com/eaaCatalog/foreignKeyMapping.html) and [value object](https://martinfowler.com/eaaCatalog/valueObject.html).
+
+### Data architecture
+
+The data model is composed of the following core entities:
+
+|Entity|Description|
+|-|-|
+|[Profile](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/models/profile.ts)|Represents the profile of a citizen (i.e. his preferences)|
+|[Message](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/models/message.ts)|Represents a message sent to a citizen by a service owned by an Organization|
+|[Notification](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/models/notification.ts)|Represents a notification to the citizen, triggered by a Message|
+|[Organization](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/models/organization.ts)|Represents an Organization whos services call the APIs to sent messages or access profiles|
```

Comment:
mayeb better: "Represents the Organization that send the messages or access citizens profile using the APIs"

Reply:
👍